### ISSUE TITLE
wrong url preview when refreshing the page in #admin and #announce workspace

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23665]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1- Create a workspace.
2- Go to the admin or announce page.
3- Type three URLs in the chat, one by one.
4- Refresh the page.
5- The last URL preview is shown for all three URLs.

## Expected Result: 
URLs should display its own respective preview
## Actual Result: 
The last URL preview is shown for all  URLs

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.46-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

[preview-bug.webm](https://github.com/Expensify/App/assets/43996225/fe2e5af0-e478-4369-8f66-f970a6e1574a)

https://github.com/Expensify/App/assets/43996225/cae00d77-4df0-4abf-8c45-7ef2a350ce05

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hichamcc
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690286102165589

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by daordonez11** [comment]
Yup tried to create a proposal but this is a backend issue! the `ExpandURLPreview` command is sending the wrong `linkMetadata `to the specific report action e.g. I have this action with google.com:
``` json
{
  "person": [
    {
      "type": "TEXT",
      "style": "strong",
      "text": "Daniel Ordóñez"
    }
  ],
  "actorAccountID": 14932266,
  "message": [
    {
      "type": "COMMENT",
      "html": "<a href=\"https://google.com\" target=\"_blank\" rel=\"noreferrer noopener\">google.com</a>",
      "text": "google.com",
      "isEdited": false,
      "whisperedTo": [],
      "isDeletedParentAction": false,
      "reactions": []
    }
  ],
  "originalMessage": {
    "html": "<a href=\"https://google.com\" target=\"_blank\" rel=\"noreferrer noopener\">google.com</a>",
    "lastModified": "2023-07-26 21:02:56.572"
  },
  "avatar": "https://d1wpcgnaa73g0y.cloudfront.net/924f64f3412dddab2627463cc46717d7fb6fd018_128.jpeg",
  "created": "2023-07-26 21:02:56.572",
  "timestamp": 1690405376,
  "reportActionTimestamp": 1690405376572,
  "automatic": false,
  "actionName": "ADDCOMMENT",
  "shouldShow": true,
  "reportActionID": "8815357875207977280",
  "lastModified": "2023-07-26 21:02:56.572",
  "whisperedToAccountIDs": []
}
```

Later when `ExpandUrlPreview` is called you can see the same action is updated with the wrong linkMetadata in the server response:
``` json
{
  "onyxMethod": "merge",
  "key": "reportActions_8213005085178512",
  "value": {
    "8815357875207977280": {
      "linkMetadata": [
        {
          "title": "Facebook - log in or sign up",
          "description": "Log into Facebook to start sharing and connecting with your friends, family, and people you know.",
          "lang": "en",
          "author": null,
          "publisher": "Facebook",
          "image": {
            "url": "https://www.facebook.com/images/fb_icon_325x325.png",
            "type": "png",
            "size": 2613,
            "height": 325,
            "width": 325,
            "size_pretty": "2.61 kB"
          },
          "date": "2023-07-26T16:52:30.000Z",
          "url": "https://www.facebook.com/",
          "logo": {
            "url": "https://static.xx.fbcdn.net/rsrc.php/yb/r/hLRJ1GG_y0J.ico",
            "type": "ico",
            "size": 4286,
            "height": 32,
            "width": 32,
            "size_pretty": "4.29 kB"
          }
        }
      ]
    }
  }
}
```

So @conorpendergrast  I think this should be internal!

cc @mountiny

**Comment by mountiny** [comment]
@conorpendergrast once you get to this, I think we should assign Daniel and give them partial reward for help narrowing this down

**Comment by mountiny** [comment]
I think standard for partial rewards if 25% and we should wait til this is addressed on backend before paying out, thanks!

**Comment by conorpendergrast** [comment]
Sounds good. @daordonez11 could you apply to [this job](https://www.upwork.com/jobs/~01bdb84617026e5d37) please?

**Comment by daordonez11** [comment]
For sure! applied @conorpendergrast let me know if I can help with anything during the backend fix

**Comment by conorpendergrast** [comment]
Hired!

**Comment by techievivek** [comment]
Interesting, for some reason this only happens on #admin/#announce room but not on a DM.

**Comment by techievivek** [comment]
Ok, just debugged this on the backend and it looks like we are fetching the wrong HTML content for a given reportActionID here https://github.com/Expensify/Web-Expensify/blob/b5f7810a4df1fd71fa94c80938a8f094d2b8c856/lib/ReportAPI.php#L354

**Comment by conorpendergrast** [comment]
Looks like the PR was sent for re-review yesterday!

**Comment by conorpendergrast** [comment]
Seems like the PR is going back to the drawing board!

**Comment by techievivek** [comment]
@conorpendergrast  This was merged last week, so we should be good to issue the reporting bonus and close the issue.

**Comment by conorpendergrast** [comment]
@hichamcc Please apply to [this job](https://www.upwork.com/jobs/~01bdb84617026e5d37) for the reporting bonus, thanks! 

@daordonez11 Paid!

**Comment by hichamcc** [comment]
@conorpendergrast applied

**Comment by techievivek** [comment]
Are we done with the payment here @conorpendergrast?

**Comment by hichamcc** [comment]
@techievivek  I haven't received the reporting bonus yet

**Comment by conorpendergrast** [comment]
Ah, offer sent, sorry for the delay @hichamcc

**Comment by hichamcc** [comment]
@conorpendergrast accepted , thank you

**Comment by conorpendergrast** [comment]
Paid, thanks for that. Closing out!
